### PR TITLE
feat(auth): Add Email Privacy support in Project and Tenant config

### PR DESCRIPTION
**Pull request by pragatimodi**

### CODE REVIEW

**Review by renkelvin — APPROVED** [review]

File: src/auth/tenant.ts [hunk 1]
```diff
@@ -299,6 +313,10 @@ export class Tenant {
       // This will throw an error if invalid.
       PasswordPolicyAuthConfig.buildServerRequest(request.passwordPolicyConfig);
     }
+    // Validate Email Privacy Config if provided.
+    if (typeof request.emailPrivacyConfig != 'undefined') {
```
Comment (Xiaoshouzi-gh) [inline src/auth/tenant.ts:317]: !==

File: test/integration/auth.spec.ts [hunk 2]
```diff
@@ -1912,6 +1932,7 @@ describe('admin.auth', () => {
         multiFactorConfig: deepCopy(expectedUpdatedTenant.multiFactorConfig),
         testPhoneNumbers: deepCopy(expectedUpdatedTenant.testPhoneNumbers),
         recaptchaConfig: deepCopy(expectedUpdatedTenant.recaptchaConfig),
+        emailPrivacyConfig: { enableImprovedEmailPrivacy: false },
```
Comment (Xiaoshouzi-gh) [inline test/integration/auth.spec.ts:1935]: can you add this to expectedUpdatedTenant?

File: test/integration/auth.spec.ts [hunk 3]
```diff
@@ -1923,6 +1944,9 @@ describe('admin.auth', () => {
         testPhoneNumbers: null,
         smsRegionConfig: deepCopy(expectedUpdatedTenant2.smsRegionConfig),
         recaptchaConfig: deepCopy(expectedUpdatedTenant2.recaptchaConfig),
+        emailPrivacyConfig: {
```
Comment (Xiaoshouzi-gh) [inline test/integration/auth.spec.ts:1947]: same here, please add an object to expectedUpdatedTenant2 and all other occurrence?

**Review by Xiaoshouzi-gh — CHANGES_REQUESTED** [review]

File: test/integration/auth.spec.ts (hunk 2 above)
Comment (pragatimodi) [reply test/integration/auth.spec.ts:1935]: `expectedUpdatedTenant` will yield `emailPrivacyConfig: {}` if the `enableImprovedPrivacy` variable is set to false. So we can't add this to that else the test will fail.

File: test/integration/auth.spec.ts (hunk 3 above)
Comment (pragatimodi) [reply test/integration/auth.spec.ts:1947]: Explained in the other comment.

File: src/auth/auth-config.ts [hunk 4]
```diff
@@ -2281,3 +2281,50 @@ export interface CustomStrengthOptionsAuthServerConfig {
   minPasswordLength?: number;
   maxPasswordLength?: number;
 }
+
+/**
+ * The configuration for the email privacy on the project or tenant.
+*/
+export interface EmailPrivacyConfig {
+  /**
+   * Variable indicating email privacy enabled of not.
```
Comment (kevinthecheung) [inline src/auth/auth-config.ts:2290]:
```suggestion
   * Whether enhanced email privacy is enabled.
```

**Review by kevinthecheung — APPROVED** [review]

**Review by Xiaoshouzi-gh — APPROVED** [review]

**Review by lahirumaramba — APPROVED** [review]
Thanks! ~Please get the docs reviewed by a TW before merging~
Just noticed Kevin's reviews. Thanks

**Comment by pragatimodi** [comment]
Hey @lahirumaramba, looks like the `remoteConfig` integration tests are failing. But they aren't related to this change, should we go ahead and merge this PR then?

**Pull request merged**